### PR TITLE
Add `update_marketing_version` workflow

### PR DESCRIPTION
**Pull request by peagasilva**
Closes [TMOB-938] and allows https://github.com/WeTransfer/Mule/pull/1817.

This PR adds a workflow to update the marketing version. An example of the PR it opens can be seen [here](https://github.com/WeTransfer/Mule/pull/1820).

To update the marketing version, you can simply run the following on Slack: `/mule branch:develop | workflow:update_marketing_version | ENV[NEW_VERSION]:{new_version}`.

[TMOB-938]: https://wetransfer.atlassian.net/browse/TMOB-938?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by wetransferplatform** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  View more details on <a href="https://app.bitrise.io/build/0b8e0080-f7c7-4842-bd19-9a7cafd37713" target="_blank">Bitrise</a>
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>WeTransferPRLinterTests: Executed 18 tests (0 failed, 0 retried, 0 skipped) in 15.241 seconds</td>
    </tr>
  </tbody>
</table>

## Code Coverage Report
| Name | Coverage ||
| --- | --- | --- |
WeTransferPRLinter | 92.19% | ✅
WeTransferPRLinter | 0.00% | ⚠️

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against b23957d0739f7a8bdc6bf43b48e00cedc4484025
</p>

File: bitrise.yml [hunk 1]
```diff
@@ -67,3 +67,56 @@ workflows:
...
+            fi
+
+            BRANCH_NAME="build/new-version/${NEW_VERSION}"
+            git checkout -b ${BRANCH_NAME}
+
+            CURRENT_VERSION=$(sed -n '/MARKETING_VERSION/{s/MARKETING_VERSION = //;s/;//;s/^[[:space:]]*//;p;q;}' ./Transfer.xcodeproj/project.pbxproj)
+
+            sed -i '' -e 's/MARKETING_VERSION \= [^\;]*\;/MARKETING_VERSION = '"${NEW_VERSION}"';/' ./Transfer.xcodeproj/project.pbxproj
+
+            # git fetch origin raph/workflow/test
```
Comment (peagasilva) [inline bitrise.yml:93]:
```suggestion
```

File: bitrise.yml [hunk 2]
```diff
@@ -67,3 +67,56 @@ workflows:
...
+            git commit -am ${NEW_VERSION_TEXT}
+
+            git push --set-upstream origin ${BRANCH_NAME} --force
+
+            if ! [ $? -eq 0 ]; then
+                echo "Pushing ${BRANCH_NAME} to the repo failed."
+                exit 1
+            fi
+
+            # For more info, please refer to `Assign PR author`
```
Comment (peagasilva) [inline bitrise.yml:105]:
```suggestion
            # For more info on how the repository name is parsed, please refer to `Assign PR author`
```

**Comment by peagasilva** [comment]
The branch push doesn't seem to be working as it should. Moving it to draft for now.

File: bitrise.yml [hunk 3]
```diff
@@ -67,3 +67,54 @@ workflows:
...
+                -d "{\"title\":\"${NEW_VERSION_TEXT}\",\"body\": \"Updated from \`${CURRENT_VERSION}\` to \`${NEW_VERSION}\`\",\"head\":\"${BRANCH_NAME}\",\"base\":\"develop\"}" \
+                &> /dev/null
+
+            if ! [ $? -eq 0 ]; then
+                echo "Pull request creation for ${BRANCH_NAME} failed. Deleting the branch..."
+
+                git push origin --delete ${BRANCH_NAME}
+
+                exit 1
+            fi
```
Comment (AvdLee) [inline bitrise.yml:120]: This is great, you even took care of handling failures correctly! Can you move it into a separate file? It's nicer to have these script files so we could eventually easier migrate to Xcode Cloud, for example.

**Review by AvdLee — CHANGES_REQUESTED** [review]

**Comment by AvdLee** [comment]
This is great @peagasilva! What do you think about making it run the TestFlight delivery right after? It's something we need either way, since we want to get into TestFlight beta review as soon as possible, so we can deliver new QA builds asap as well.

File: bitrise.yml (hunk 3 above)
Comment (peagasilva) [reply bitrise.yml]: Addressed in https://github.com/WeTransfer/WeTransfer-iOS-CI/pull/175/commits/32b45edede50caf57806905123df7ccda92bacc0.

File: bitrise.yml [hunk 4]
```diff
@@ -67,3 +67,18 @@ workflows:
             #!/bin/bash
             set -ex
             bitrise run --config ./Bitrise/tag_releasing_bitrise.yml "${BITRISE_TRIGGERED_WORKFLOW_ID}"
+  update_marketing_version:
+    steps:
+    # This step expects `activate-ssh-key` and `git-clone` to be run beforehand
```
Comment (peagasilva) [inline bitrise.yml:72]: This spare us a few seconds ⏳

**Comment by peagasilva** [comment]
@AvdLee your suggestions were addressed and the PR is ready for another pass. Thanks!

**Review by AvdLee — APPROVED** [review]
Can you rename the folder `bin` to `scripts`? We use it in other projects like Coyote and Mule, which would make it more consistent. Other than that, it looks good!

**Pull request merged**